clients/src/main/resources/common/message/ConsumerGroupHeartbeatRequest.json

```diff
@@ -0,0 +1,66 @@
+// limitations under the License.
+
+{
+  "apiKey": 68,
+  "type": "request",
+  "apiStability": "evolving",
+  "listeners": ["zkBroker", "broker"],
+  "name": "ConsumerGroupHeartbeatRequest",
+  "validVersions": "0",
+  "flexibleVersions": "0+",
+  "fields": [
+    { "name": "GroupId", "type": "string", "versions": "0+", "entityType": "groupId",
+      "about": "The group identifier." },
+    { "name": "MemberId", "type": "string", "versions": "0+",
+      "about": "The member id generated by the coordinator. The member id must be kept during the entire lifetime of the member." },
+    { "name": "MemberEpoch", "type": "int32", "versions": "0+",
+      "about": "The current member epoch; 0 to join the group; -1 to leave the group; -2 to indicate that the static member will rejoin." },
+    { "name": "InstanceId", "type": "string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if not provided or if it didn't change since the last heartbeat; the instance Id otherwise." },
+    { "name": "RackId", "type": "string", "versions": "0+",  "nullableVersions": "0+", "default": "null",
+      "about": "null if not provided or if it didn't change since the last heartbeat; the rack ID of consumer otherwise." },
+    { "name": "RebalanceTimeoutMs", "type": "int32", "versions": "0+", "default": -1,
+      "about": "-1 if it didn't chance since the last heartbeat; the maximum time in milliseconds that the coordinator will wait on the member to revoke its partitions otherwise." },
+    { "name": "SubscribedTopicNames", "type": "[]string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if it didn't change since the last heartbeat; the subscribed topic names otherwise." },
+    { "name": "SubscribedTopicRegex", "type": "string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if it didn't change since the last heartbeat; the subscribed topic regex otherwise" },
+    { "name": "ServerAssignor", "type": "string", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if not used or if it didn't change since the last heartbeat; the server side assignor to use otherwise." },
+    { "name": "ClientAssignors", "type": "[]Assignor", "versions": "0+", "nullableVersions": "0+", "default": "null",
+      "about": "null if not used or if it didn't change since the last heartbeat; the list of client-side assignors otherwise.", "fields": [
+      { "name": "Name", "type": "string", "versions": "0+",
+        "about": "The name of the assignor." },
+      { "name": "MinimumVersion", "type": "int16", "versions": "0+",
+        "about": "The minimum supported version for the metadata." },
+      { "name": "MaximumVersion", "type": "int16", "versions": "0+",
+        "about": "The maximum supported version for the metadata." },
+      { "name": "Reason", "type": "int8", "versions": "0+",
```

Comment:
can you provide an example of what we store in the byte?

Reply:
It just occurs to me: shall we reserve some numbers for AK client usage? I know that in the KIP we did not do so, for Streams e.g. it can start with reason 0 for its own meaning. But thinking about this a bit more I'm wondering if some common reasons that are used by consumer clients directly should be reserved? E.g. re-discover coordinator after disconnect, subscription changes, etc.